src/schemas.jl

```diff
@@ -165,6 +189,25 @@ written via [`Legolas.write`](@ref).
 """
 identifier(sv::SchemaVersion) = throw(UnknownSchemaVersionError(sv))
 
+"""
+    Legolas.schema_provider(::SchemaVersion)
+
+Returns a NamedTuple with keys `name` and `version`. The name is a `Symbol` corresponding to the package which defines the schema version, if known; otherwise `nothing`. Likewise the `version` is a `VersionNumber` or `nothing`.
+"""
+schema_provider(::SchemaVersion) = (; name=nothing, version=nothing)
+# shadow `pkgversion` so we don't fail on pre-1.9
+pkgversion(m::Module) = isdefined(Base, :pkgversion) ? Base.pkgversion(m) : nothing
+
+# Used in the implementation of `schema_provider`.
+function defining_package_version(m::Module)
+    rootmodule = Base.moduleroot(m)
+    # Check if this module was defined in a package.
+    # If not, return `nothing`
+    path = pathof(rootmodule)
+    path === nothing && return (; name=nothing, version=nothing)
```

Comment:
Comment above makes it sound like this should be:
```suggestion
    path === nothing && return nothing
```

Reply:
needs to be consistent with the return value described in the docstring of `schema_provider` though, which is as it is here...

Reply:
fixed the comment